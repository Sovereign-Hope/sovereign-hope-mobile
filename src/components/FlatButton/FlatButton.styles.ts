import { StyleSheet, TextStyle, ViewStyle } from "react-native";
import { elementSize, elevation, radius, spacing } from "src/style/layout";
import { Theme } from "@react-navigation/native";
import { colors } from "src/style/colors";

type Props = {
  theme: Theme;
};

interface Style {
  button: ViewStyle;
  icon: ViewStyle;
  text: TextStyle;
}

export const styles = ({ theme }: Props): Style =>
  StyleSheet.create({
    button: {
      flexDirection: "row",
      alignItems: "center",
      justifyContent: "center",
      paddingVertical: spacing.medium,
      paddingHorizontal: spacing.large,
      minHeight: elementSize.small,
      borderRadius: radius.medium,
      elevation: elevation.small,
      backgroundColor: theme.dark ? colors.brand : colors.darkGrey,
    },
    icon: {
      marginRight: spacing.small,
    },
    text: {
      color: colors.white,
      flex: 1,
      textAlign: "center",
    },
  });
