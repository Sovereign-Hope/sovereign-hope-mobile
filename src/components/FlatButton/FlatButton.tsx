import * as React from "react";
import { Pressable, Text, ViewStyle } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { elementSize } from "src/style/layout";
import { text } from "src/style/colors";
import { useTheme } from "@react-navigation/native";
import { styles } from "./FlatButton.styles";

interface ButtonProps {
  title: string;
  disabled?: boolean;
  onPress: () => void;
  style?: ViewStyle;
  icon?: keyof typeof Ionicons.glyphMap;
}

export const FlatButton: React.FunctionComponent<ButtonProps> = ({
  title,
  disabled,
  onPress,
  style,
  icon,
}: ButtonProps) => {
  const theme = useTheme();
  const themedStyles = styles({ theme });

  return (
    <Pressable
      style={({ pressed }) => ({
        ...themedStyles.button,
        ...style,
        opacity: pressed || disabled ? 0.6 : 1,
      })}
      onPress={onPress}
      disabled={disabled}
    >
      {icon && (
        <Ionicons
          name={icon}
          size={elementSize.tiny}
          color={text.light}
          style={themedStyles.icon}
        />
      )}
      <Text style={themedStyles.text}>{title}</Text>
    </Pressable>
  );
};

FlatButton.defaultProps = {
  style: {},
  disabled: false,
  icon: undefined,
};
