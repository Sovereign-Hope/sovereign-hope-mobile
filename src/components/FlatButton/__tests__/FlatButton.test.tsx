import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { FlatButton } from "../FlatButton";

describe("FlatButton", () => {
  it("should render", () => {
    const wrapper = render(<FlatButton title="test" onPress={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the title", () => {
    const titleText = "test";
    const wrapper = render(<FlatButton title={titleText} onPress={() => {}} />);
    const title = wrapper.getByText(titleText);
    expect(title).toBeTruthy();
  });

  it("should render an icon", () => {
    const wrapper = render(
      <FlatButton title="test" icon="mail" onPress={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger the onPress action", () => {
    const mock = jest.fn();
    const wrapper = render(<FlatButton title="test" onPress={mock} />);
    fireEvent.press(wrapper.getByText("test"));
    expect(mock).toHaveBeenCalled();
  });
});
