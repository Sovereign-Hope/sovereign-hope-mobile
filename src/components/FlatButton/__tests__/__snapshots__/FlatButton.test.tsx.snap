// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatButton should render 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#333333",
      "borderRadius": 10,
      "elevation": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 44,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "flex": 1,
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
</View>
`;

exports[`FlatButton should render an icon 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#333333",
      "borderRadius": 10,
      "elevation": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 44,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 22,
        },
        Object {
          "marginRight": 4,
        },
        Object {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    Ôèú
  </Text>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "flex": 1,
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
</View>
`;
